documentation/proposals/Proposal - Axis Input Devices.md

```diff
@@ -0,0 +1,624 @@
+/// It has specifications that must be followed to be considered valid
+/// </summary>
+[Flags]
+public enum AxisGroupType : uint
+{
+    Unknown = 0u,
+
+    // an ordering convention for family axes is left/right/bottom/top/back/forward - "negative" to "positive" values
+
+    // (West, East, South, North) / (Left, Right, Down, Up) / (X, B, A, Y) - xbox / (square, circle, cross, triangle) - playstation
+    // 4 independent axes using the ordering convention above
+    FourFaceButtons = 1u << 1,
+
+    // Specified in addition to FourFaceButtons due to the constraints often placed on DPad axes from a hardware level
+    // 2 Axes, (X, Y) (-1, 1)
+    // made of 4 physical axes, (X-, X+, Y-, Y+)
+    DPad = 1u << 2 | FourFaceButtons,
+    
+    /// <summary>
+    /// 1D joystick, 1 logical axis, (-1, 1)
+    /// made of 2 physical axes, (X-, X+)
+    /// More physically resembling a "lever" than a joystick
+    /// </summary>
+    Joystick1D = 1u << 3 | DPad,
+
+    /// <summary>
+    /// 2D joystick, 2 logical axes, (X, Y) (-1, 1)
+    /// made of 4 physical axes, (X-, X+, Y-, Y+)
+    /// If constructed with 5 axes, the 5th axis is the "pressure" axis for a pointer
+    /// More abstractly, this is a Position2D that returns to 0,0 when released
+    /// </summary>
+    Joystick2D = 1u << 4 | Position2D | DPad,
+
+    /// <summary>
+    /// 3D joystick
+    /// 3 logical Axes (X, Y, Z), (-1, 1)
+    /// made of 6 physical axes, (X-, X+, Y-, Y+, Z-, Z+)
+    /// If constructed with 7 axes, the 7th axis is the "pressure" axis for a pointer
+    /// Can be used for analog stick with a press-in button, or a 3D mouse, etc
+    /// More abstractly, this is a Position3D that returns to 0,0,0 when released
+    /// </summary>
+    Joystick3D = 1u << 5 | Position3D,
+    
+    /// <summary>
+    /// 1D touch surface or tracker
+    /// 1 logical axis, (-1, 1)
+    /// made of 2 physical axes, (-X, +X)
+    /// Physically, this can be a represented as a touch strip
+    /// </summary>
+    Position1D = 1u << 6,
+
+    /// <summary>
+    /// 2D touch surface or tracker
+    /// 2 logical axes, (X, Y) (-1, 1)
+    /// made of 4 physical axes (-X, +X, -Y, +Y)
+    /// If constructed with 5 axes, the 5th axis is the "pressure" axis for a pointer
+    /// </summary>
+    Position2D = 1u << 7,
+
+    /// <summary>
+    /// 3D touch surface or tracker, 3 logical axes, (X, Y, Z)
+    /// made of 6 physical axes (-X, +X, -Y, +Y, -Z, +Z)
+    /// If constructed with 7 axes, the 7th axis is the "pressure" axis for a pointer
+    /// </summary>
+    Position3D = 1u << 8,
+
+    // should this be 6 and 8 axes, similar to how a 2d joystick is defined by 4 axes?
+    RotationEuler = 1u << 9, // gyroscope, gyroscope + magnetomete, VR peripheral rotation - requires 3 axes, XYZ order
+
+    RotationQuaternion =
+        1u << 10, // gyroscope, gyroscope + magnetomete, VR peripheral rotation - requires 4 axes, XYZW order

```

Comment:
Aha you've answered my questions! Will keep them posted to provide insight into the interpretation of this doc but will mark them as resolved.